### PR TITLE
AG-10782 Fix zoom range buttons not triggering update after navigator drag

### PR DESCRIPTION
**Pull request by lsjroberts**
https://ag-grid.atlassian.net/browse/AG-10782

### CODE REVIEW

File: packages/ag-charts-enterprise/src/features/zoom/zoomRange.ts [hunk 1]
```diff
@@ -35,6 +38,35 @@ export class ZoomRange {
...
+
+    public extendWith(fn: (end: Date | number) => Date | number) {
+        if (!this.domain) return;
+
+        const end = this.domain.at(-1);
+        if (end == null) return;
+
+        this.dirty = true;
+
+        this.end = end;
```
Comment (alantreadway) [inline packages/ag-charts-enterprise/src/features/zoom/zoomRange.ts:51]: Discussed offline: Maybe add a comment that `this.dirty` maybe reset as a side-effect of setting these fields?

File: packages/ag-charts-enterprise/src/features/zoom/zoomRange.ts [hunk 2]
```diff
@@ -35,6 +38,35 @@ export class ZoomRange {
...
+
+        if (this.dirty) this.onChange?.(this.getRange());
+    }
+
+    public extendAll() {
+        if (!this.domain) return;
+
+        this.dirty = true;
+
+        this.start = this.domain[0];
```
Comment (alantreadway) [inline packages/ag-charts-enterprise/src/features/zoom/zoomRange.ts]: Ditto

File: packages/ag-charts-enterprise/src/features/zoom/zoomRange.ts (hunk 1 above)
Comment (lsjroberts) [reply packages/ag-charts-enterprise/src/features/zoom/zoomRange.ts:51]: Changed it to check the values have changed directly and not use side-effects.

**Review by alantreadway — APPROVED** [review]
LGTM

**Pull request merged**